tests/x11/mac80211_hwsim.pm

```diff
@@ -0,0 +1,219 @@
+    # select it
+    assert_and_click 'gnome_widget-choose_network-click';
+    # and click on 'connect'
+    assert_and_click 'gnome_widget-connect-click';
+    $self->enter_NM_credentials;
+}
+
+sub enter_NM_credentials {
+    my $self = shift;
+    # we expect here to enter our credentials for this wireless network
+    assert_screen 'network_manager-wpa2_authentication';
+
+    # When we land at this dialog, nothing has focus.
+    # The first tab results in focusing the first input field.
+    # But since we want the dropdown field one above, we have
+    # to go up by pressing "Shift+Tab".
+
+    send_key 'tab';
+    hold_key 'shift';
+    send_key 'tab';
+    release_key 'shift';
+
+    # then we open the dropdown
+    send_key 'spc';
+    # and select 'Protected EAP (PEAP)'
+    send_key_until_needlematch('network_manager-peap_selected', 'up');
+    send_key 'ret';
+
+    # enter anonymous identity
+    type_string 'franz.nord@example.com';
+
+    # select 'No CA certificate needed'
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'spc';
+    
+    # 2x tab, franz.nord@example.com
+    send_key 'tab';
+    send_key 'tab';
+    send_key 'tab';
+    type_string 'franz.nord@example.com';
+    send_key 'tab';
+    type_string 'nots3cr3t';
+
+    # aat 'connect'
+    assert_and_click 'network_manager-connect-click';
+}
+
+sub generate_certs {
+    assert_script_run 'mkdir -p wpa_enterprise_certificates/{CA,server}';
+    assert_script_run 'cd wpa_enterprise_certificates';
+
+    type_string "# generate private keys\n";
+    assert_script_run 'openssl genrsa -out CA/CA.key 4096';
+    assert_script_run 'openssl genrsa -out server/server.key 4096';
+    save_screenshot;
+
+    type_string "# generate certificate for CA\n";
+    assert_script_run 'openssl req -x509 -new -nodes -key CA/CA.key -sha256 -days 3650 -out CA/CA.crt -subj "/"';
+
+    type_string "# generate certificate signing request for server\n";
+    assert_script_run 'openssl req -new -key server/server.key -out server/server.csr -subj "/"';
+    save_screenshot;
+
+    type_string "# sign csr with the key/cert from the CA\n";
+    assert_script_run 'openssl x509 -req -in server/server.csr -CA CA/CA.crt -CAkey CA/CA.key -CAcreateserial -out server/server.crt -days 3650 -sha256';
+    save_screenshot;
+}
+
+sub configure_hostapd {
+    type_string "# configure hostapd\n";
+    assert_script_run 'wget -O /etc/hostapd.conf ' . data_url('hostapd_wpa2-enterprise.conf');
+
+    type_string "# create wpa2 enterprise user\n";
+    assert_script_run 'echo \"franz.nord@example.com\" PEAP >> /etc/hostapd.eap_user';
+    assert_script_run 'echo \"franz.nord@example.com\" MSCHAPV2 \"nots3cr3t\" [2]>> /etc/hostapd.eap_user';
+}
+
+sub configure_system {
+    my $self = shift;
+    y2x11test::launch_yast2_module_x11;
+    $self->yast_search('Network Settings');
+
+    # assert_and_click 'yast2-networksettings'
```

Comment:
some of these comments are a bit to verbose and don't add information. Looks like left-overs from initial pseudo code? Better delete some comments now.